src/selectors/prevThought.ts

```diff
@@ -0,0 +1,124 @@
+  const effectivePath = isInContextView ? simplifyPath(state, path) : path
+  const id = head(effectivePath)
+
+  // Only traverse children if the thought or its parent is expanded
+  if (!state.expanded[hashPath(effectivePath)]) return path
+
+  const children = getChildren(state, id)
+  // If no children, return current path
+  if (children.length === 0) return path
+
+  const lastChild = children[children.length - 1]
+  return lastVisibleDescendant(state, appendToPath(path, lastChild.id))
+}
+
+/** Gets the last context in a context view. */
+const lastContext = (state: State, path: Path): Path | null => {
+  const thought = getThoughtById(state, head(path))
+  const contexts = getContextsSortedAndRanked(state, thought.value).filter(
+    context => state.showHiddenThoughts || isVisible(state, getThoughtById(state, context.id)),
+  )
+
+  // if context view is empty or no visible contexts, return null
+  if (contexts.length === 0) return null
+
+  // get the last visible context
+  const lastContext = getThoughtById(state, contexts[contexts.length - 1]?.id)
+  if (!state.expanded[hashPath(appendToPath(path, lastContext.parentId))])
+    return appendToPath(path, lastContext.parentId)
+  return lastVisibleDescendant(state, appendToPath(path, lastContext.parentId))
+}
+
+/**
+ * Gets the last visible descendant of the previous sibling.
+ * If the previous sibling is hidden, recursively tries the next previous sibling.
+ */
+const getLastVisibleDescendantOfPreviousSibling = (
+  state: State,
+  path: Path,
+  currentSibling: Path | null = null,
+): Path | null => {
+  const currentPath = currentSibling || path
+  const prevSiblingThought = prevSibling(state, currentPath)
+  if (!prevSiblingThought) return null
+  // Skip hidden thoughts only when showHiddenThoughts is false
+  const isHidden = !state.showHiddenThoughts && !isVisible(state, prevSiblingThought)
+  if (!isHidden) {
+    const prevSiblingPath = appendToPath(parentOf(path), prevSiblingThought.id)
+    return lastVisibleDescendant(state, prevSiblingPath)
+  }
+
+  const prevPath = appendToPath(parentOf(path), prevSiblingThought.id)
+  return getLastVisibleDescendantOfPreviousSibling(state, path, prevPath)
+}
+
+/** Recursively checks if any parent up to root is a context view root. */
+const findContextViewRoot = (state: State, path: Path): Path | null => {
+  if (isRoot(path)) return null
+  if (isContextViewActive(state, path)) return path
+  return findContextViewRoot(state, rootedParentOf(state, path))
+}
+
+/** Gets the previous thought in visual order. */
+const prevThought = (state: State, path?: Path): Path | null => {
+  if (!path) return null
+
+  const pathParent = rootedParentOf(state, path)
+  const isRootLevel = isRoot(pathParent)
+
+  // Check if we're in a context view
+  const inContextView = isContextViewActive(state, pathParent)
+  const hasActiveContextViews = Object.keys(state.contextViews).length > 0
```

Comment:
This checks if *any* context view is active, even if it is far away from the path. You would need to devise a more specific condition.